process/hyperBlocksPool.go

```diff
@@ -0,0 +1,125 @@
+package process
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-ws-connector-firehose-go/data"
+	"github.com/multiversx/mx-chain-ws-connector-firehose-go/data/hyperOutportBlocks"
+)
+
+type hyperOutportBlocksPool struct {
+	marshaller marshal.Marshalizer
+	dataPool   DataPool
+}
+
+// NewHyperOutportBlocksPool will create a new instance of hyper outport blocks pool
+func NewHyperOutportBlocksPool(
+	dataPool DataPool,
+	marshaller marshal.Marshalizer,
+) (*hyperOutportBlocksPool, error) {
+	if check.IfNil(dataPool) {
+		return nil, ErrNilDataPool
+	}
+	if check.IfNil(marshaller) {
+		return nil, ErrNilMarshaller
+	}
+
+	return &hyperOutportBlocksPool{
+		dataPool:   dataPool,
+		marshaller: marshaller,
+	}, nil
+}
+
+// UpdateMetaState will triiger meta state update from base data pool
+func (bp *hyperOutportBlocksPool) UpdateMetaState(checkpoint *data.BlockCheckpoint) error {
+	return bp.dataPool.UpdateMetaState(checkpoint)
+}
+
+// Get will trigger data pool get operation
+func (bp *hyperOutportBlocksPool) Get(key []byte) ([]byte, error) {
+	return bp.dataPool.Get(key)
+}
+
+// PutMetaBlock will put the provided meta outport block data to the pool
+func (bp *hyperOutportBlocksPool) PutMetaBlock(hash []byte, outportBlock *hyperOutportBlocks.MetaOutportBlock) error {
+	shardID := outportBlock.ShardID
+	currentIndex := outportBlock.BlockData.Header.GetRound()
+
+	previousIndex := outportBlock.HighestFinalBlockNonce
+	isSuccesiveIndex := currentIndex >= previousIndex
```

Comment:
```suggestion
	isSuccessiveIndex := currentIndex >= previousIndex
```

also, should the check be something like `currentIndex == previousIndex+1`?

otherwise I'd suggest renaming `isSuccessiveIndex` to `isHigherIndex`

Reply:
renamed to `isHigherIndex`